src/analysis/processing/qgsalgorithmdownloadvectortiles.cpp

```diff
@@ -0,0 +1,225 @@
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmdownloadvectortiles.h"
+
+#include "qgsvectortilelayer.h"
+#include "qgsvectortiledataprovider.h"
+#include "qgsziputils.h"
+
+///@cond PRIVATE
+
+class SetStylePostProcessor : public QgsProcessingLayerPostProcessorInterface
+{
+  public:
+
+    SetStylePostProcessor( QgsVectorTileLayer *inputLayer )
+      : mLayer( inputLayer )
+    {}
+
+    void postProcessLayer( QgsMapLayer *layer, QgsProcessingContext &, QgsProcessingFeedback * ) override
+    {
+      if ( QgsVectorTileLayer *tileLayer = qobject_cast< QgsVectorTileLayer * >( layer ) )
+      {
+        QDomDocument doc( QStringLiteral( "qgis" ) );
+        QString errorMsg;
+        mLayer->exportNamedStyle( doc, errorMsg );
+        tileLayer->importNamedStyle( doc, errorMsg );
+        tileLayer->triggerRepaint();
+      }
+    }
+
+  private:
+
+    QgsVectorTileLayer *mLayer;
+};
+
+QString QgsDownloadVectorTilesAlgorithm::name() const
+{
+  return QStringLiteral( "downloadvectortiles" );
+}
+
+QString QgsDownloadVectorTilesAlgorithm::displayName() const
+{
+  return QObject::tr( "Download vector tiles" );
+}
+
+QStringList QgsDownloadVectorTilesAlgorithm::tags() const
+{
+  return QObject::tr( "vector,split,field,unique" ).split( ',' );
+}
+
+QString QgsDownloadVectorTilesAlgorithm::group() const
+{
+  return QObject::tr( "Vector tiles" );
+}
+
+QString QgsDownloadVectorTilesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectortiles" );
+}
+
+QString QgsDownloadVectorTilesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Downloads vector tiles of the input vector tile layer and saves them in the local vector tile file." );
+}
+
+QgsDownloadVectorTilesAlgorithm *QgsDownloadVectorTilesAlgorithm::createInstance() const
+{
+  return new QgsDownloadVectorTilesAlgorithm();
+}
+
+void QgsDownloadVectorTilesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterMapLayer( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QVariant(), false, QList<int>() << QgsProcessing::TypeVectorTile ) );
+  addParameter( new QgsProcessingParameterExtent( QStringLiteral( "EXTENT" ), QObject::tr( "Extent" ) ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "MAX_ZOOM" ), QObject::tr( "Maximum zoom level to download" ), QgsProcessingParameterNumber::Integer, 14, false, 0 ) );
```

Comment:
For the reason given below, I'd suggest defaulting to a much more conservative maximum zoom level of say 10.

Reply:
Done.